scrabble/lib/scrabble.ex

```diff
@@ -0,0 +1,227 @@
+        val1 = Enum.at(word, 0)
+        val2 = Enum.at(word, 1)
+        val3 = Enum.at(word, 2)
+        val4 = Enum.at(word, 3)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}") #1, 2, 3, 4
+        help = wordlength(help, tempword, 4)
+        val4 = Enum.at(word, 4)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}") #1, 2, 3, 5
+        help = wordlength(help, tempword, 4)
+        val3 = Enum.at(word, 3)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}") #1, 2, 4, 5
+        help = wordlength(help, tempword, 4)
+        val2 = Enum.at(word, 2)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}") #1, 3, 4, 5
+        help = wordlength(help, tempword, 4)
+        val1 = Enum.at(word, 1)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}") #2, 3, 4, 5
+        help = wordlength(help, tempword, 4)
+        #run through all permutations of the four letter word.
+        permuteList = permute(word)
+        templist = []
+        templist = Enum.map(permuteList, fn(word) -> if(checkDict(word)) do templist ++ word else end end) |>
+        Enum.filter(& !is_nil(&1))
+        send from, {help, templist}
+    end
+  end
+
+  def sixLetterProcess() do
+    receive do
+      {help, word, from} ->
+        #run through the five letter process and add permutations of the 2, 3, 4, and 5 letter combinations to the list.
+        val1 = Enum.at(word, 0)
+        val2 = Enum.at(word, 1)
+        val3 = Enum.at(word, 2)
+        val4 = Enum.at(word, 3)
+        val5 = Enum.at(word, 4)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}#{val5}") #1, 2, 3, 4, 5
+        help = wordlength(help, tempword, 5)
+        val5 = Enum.at(word, 5)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}#{val5}") #1, 2, 3, 4, 6
+        help = wordlength(help, tempword, 5)
+        val4 = Enum.at(word, 4)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}#{val5}") #1, 2, 3, 5, 6
+        help = wordlength(help, tempword, 5)
+        val3 = Enum.at(word, 3)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}#{val5}") #1, 2, 4, 5, 6
+        help = wordlength(help, tempword, 5)
+        val2 = Enum.at(word, 2)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}#{val5}") #1, 3, 4, 5, 6
+        help = wordlength(help, tempword, 5)
+        val1 = Enum.at(word, 1)
+        tempword = String.graphemes("#{val1}#{val2}#{val3}#{val4}#{val5}") #2, 3, 4, 5, 6
+        help = wordlength(help, tempword, 5)
+        #run through all permutations of the four letter word.
+        permuteList = permute(word)
+        templist = []
+        templist = Enum.map(permuteList, fn(word) -> if(checkDict(word)) do templist ++ word else end end) |>
+        Enum.filter(& !is_nil(&1))
+        send from, {help, templist}
+    end
+  end
+
+  def permute([]), do: [[]]
```

Comment:
This needs to be attributed (I'm guessing to Rosetta code (https://rosettacode.org/wiki/Permutations#Elixir)